packages/sanity/src/core/form/inputs/files/FileInput/FileActionsMenu.tsx

```diff
@@ -101,9 +113,11 @@ export function FileActionsMenu(props: Props) {
 
       <Box padding={2}>
         <Flex justify="center">
+          {/* Using a customized Popover instead of MenuButton because a MenuButton will close on click
+     and break replacing an uploaded file. */}
```

Comment:
👋 I'm a bit confused about this section 🤔 Isn't i expected behavior that the MenuButton would close when clicking it? Or was it closing when it wasn't intended (and therefore causing re-renders that were causing the focus floating all the way up the menu?)

Reply:
This comment is not concerning the focus, but is trying to explain why in this [PR](https://github.com/sanity-io/sanity/pull/4094), we replaced the MenuButton with a customized `Popover`.

Reply:
I can try to make it more clear in the comment!

Reply:
huh! the more you know! Thanks! 😄

Reply:
@ninaandal nah it's fine, I just thought that one was related to this PR it's all good for me ✨